### ISSUE TITLE
where is xception like network code which is written in original paper?

### ISSUE THREAD

**Issue by sharpstill** [zengarden/light_head_rcnn#1]
where is xception like network code which is written in original paper?

**Comment by foreverYoungGitHub** [comment]
@zengarden, Would you mind to provide the code and model of xception like network? I've try to implemented and train it with Imagenet. The best accuracy I get is around 64%. would you mind share the relative code and experience of that?

Best,

**Comment by zengarden** [comment]
@foreverYoungGitHub  I'm reproducing the efficient xception like network in TF.  Since the xception network mentioned in paper is built and trained in our internal platform, i need some time to transfer it to TF and ensure the accuracy.

**Comment by xtanitfy** [comment]
When i reproding your resaults,but it will be a runing stop randomly.I trained there days and the problems still be exists!I only use one 1080TI and i change the train_batch_per_gpu = 1 because of the consuming memory largely.

**Comment by HiKapok** [comment]
Hi, I have also tried the original Xception backbone in my re-implementation and found it worked too. And the forward time in training for backbone network is about 20ms. But it's maybe better to switch to xception-like for better acc and speed.

**Comment by MaskVulcan** [comment]
@HiKapok Could you please tell me the speed and map in test?

**Comment by HiKapok** [comment]
@MaskVulcan I didn't measure the speed in test time, in fact the nms op in TensorFlow is a little slow which is a bottleneck of the full detection pipeline. You can find the codes in my github if you are interested in. But I still recommend you wait for the official one to ensure the speed.

**Comment by HiKapok** [comment]
@MaskVulcan 74.5mAP now on pascal 2007 test.

**Comment by jiang1st** [comment]
Any update about the xception-like network?

**Comment by edgarmedina1801** [comment]
Hi, can someone provide only some insights about the Xception-like model? I think I can implement the training and some other stuff, but I'm not sure if I understood well this model. I have some questions:

- is the last layer a FC 1000?  because I don't know what is the last layer to connect the light-head R-CNN
- what does this expression (from original paper) mean?  ""Following xception design strategies, we replace all convolution layer in bottle-neck structure with channel-wise convolution. However we do not use the pre-activation design which is proposed in identity mappings [10] because of shallow network.""

- Should I replace the 7 convolutional layers(first layer and stage 1,2 and 3) for convolution layers in bottleneck structure with channel-wise convolution?

- is this the large separated convolution? from 785-792 lines
https://github.com/terrychenism/Deformable-ConvNets/blob/master/rfcn/symbols/resnet_v1_101_rfcn_light.py#L785

@HiKapok good job. 74.5mAP is Pascal VOC is inferior to Yolov2 considering Light-Head R-CNN should have a better mAP.

Thank you in advanced guys.

**Comment by YanShuo1992** [comment]
@zengarden How is the tf implementation going?  Have you finished it?  Or could you please just release the structure of the Xception?  I think lots of people are waiting for it.
Cheers

**Comment by geonseoks** [comment]
@foreverYoungGitHub Hello. Would you mind to tell me your mAP and FPS in MSCOCO using xception like backbone in Light-Head R-CNN?

**Comment by edgarmedina1801** [comment]
how many layers does xception-like backbone in Light-Head R-CNN has? they are 17 conv layers and 2 MLP?
@geonseoks how did you compute the accuracy (mAP) in COCO. It is not available in this repository. Did you use the **results** file?

**Comment by geonseoks** [comment]
@edgarmedina1801 Yes, I use ConV1~FC on Tabel 7. 

And here is my answer.

is the last layer a FC 1000? because I don't know what is the last layer to connect the light-head R-CNN
-> **I connect Stage 3 Layer to RPN's input and Stage 4 Layer to global context module's input(make stride to 1 at Stage4)**

what does this expression (from original paper) mean? ""Following xception design strategies, we replace all convolution layer in bottle-neck structure with channel-wise convolution. However we do not use the pre-activation design which is proposed in identity mappings [10] because of shallow network.""
-> **I change all 3*3 conv in Stage2~4 to channel-wise conv. And pre-activation design is described in the paper[10]. So you can understand the backbone is Xeption like ResNet. Because ResNet have bottle-neck structure. I think this backbone is similar to Mobilenet V2.**

**Comment by aggpankaj2** [comment]
@HiKapok @edgarmedina1801 @geonseoks 
I trained  rfcn--reproduce.ori_res101.coco.baseline code with coco database. but while testing i got result 
[results.txt](https://github.com/zengarden/light_head_rcnn/files/2352029/results.txt)

 Average Precision  (AP) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.049

But on coco they got 40% result. 
@edgarmedina1801  . You also got result like that  0.040 . so 0.040 is equal to 4% but how is it possible on coco. can please help me to understand result.txt file.

**Comment by karansomaiah** [comment]
@geonseoks Thanks a lot for elaborating on Xception network. I have another question if you don't mind.
When the author says channel wise convolution, is it tf.nn.depthwise_conv2d or tf.nn.separable_conv2d in tensorflow?

**Comment by geonseoks** [comment]
@karansomaiah Good question, I use separable_conv2d. I will try to use depthwise_conv2d later.

**Comment by karansomaiah** [comment]
Hey @geonseoks, I tried implementing with the information you provided but I get a lot of nan values in the loss very early in training. I even checked your code:

1. I still face the same issue
2. In line 157 where `proposal_opr` is applied, I cannot use `is_tfnms=False` since it gives me an error saying `lib_kernel.lib_fast_nms` not found. I checked further, and it seems @zengarden has removed it in the latest commit. Any help will be appriciated

Thanks in advance.

**Comment by karansomaiah** [comment]
Do you guys change the spatial_scale argument for PSAlign code in your network description for xception network definition? @geonseoks @edgarmedina1801 @HiKapok
